modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -815,4 +843,53 @@ public void shutDownSlotManager() {
         return overlappedSlots;
     }
 
+    /**
+     * Recover any messages that are persisted but not notified to the slot coordinator from killed nodes.
+     * <p>
+     * For instance if a node get killed after persisting messages but before submitting slots,
+     * until another message is published to any remaining node a new slot will not be created.
+     * Hence these messages will not get delivered until another message is published.
+     * <p>
+     * Recover mechanism here will schedule tasks for each queue so that if no message get received within the
+     * given time period that queue slot manager will create a slot and capture those messages it self.
+     */
+    public void recoverSlots() {
+
+        int threadPoolCount = 1; // Single thread is suffice for this task
+        ThreadFactory namedThreadFactory =
+                new ThreadFactoryBuilder().setNameFormat("RecoverSlotsThreadPool").build();
+        ScheduledExecutorService recoverSlotScheduler =
+                Executors.newScheduledThreadPool(threadPoolCount, namedThreadFactory);
+        queuesToRecover = slotIDMap.keySet();
+
+        recoverSlotScheduler.schedule(
```

Comment:
This is a one-shot action that becomes enabled after the given delay. [1]

[1] http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#schedule(java.lang.Runnable,%20long,%20java.util.concurrent.TimeUnit)